### DISCUSSION TITLE
What's the best practice to limit the size or number of storage that each user of my app can use?

### DISCUSSION THREAD

**Discussion by hereAlexT** [supabase/supabase#19382] [Questions] [answered]
Hey,
I want to limit the size or number that each user of my app can use. 
It's like Twitter, every user can post with 1-4 pic attached.
I also want to limit the size of storage each user of mine can use.

by reading the doc, I found that I cannot use Storage API to obtain this directly. I have to set up a Function between clients and storage to apply this feature.

I wonder is this the best practice? is there any better method to obtain this?

**Comment by probablykabari** [comment, marked as answer]
See https://supabase.com/docs/guides/storage/schema/design

Looking at the objects schema, you could achieve this with an RLS policy using owner_id and the `storage.foldername` function by having upload paths in the bucket include the post_id (or whatever you named it). Same for storage size, you can use a function and check the owner and sum all their uploads per bucket.

I have a policy like this in a similar app to restrict by channel for example:

```sql
create policy insert_own
  on storage.objects 
  for insert 
  to authenticated
  with check (
    -- folders like [channel_name]/[id]/filename
    bucket_id = 'messages' and app_public.can_access_channel(storage.foldername(name)[1], storage.foldername(name)[2])
  );
```
